central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -240,11 +240,51 @@ func (m *managerImpl) HandleScanRequestResponse(ctx context.Context, requestID s
 	return nil
 }
 
-func (m *managerImpl) ProcessRescanRequest(_ context.Context, _ interface{}) error {
-	// TODO(ROX-18091):
+func (m *managerImpl) ProcessRescanRequest(ctx context.Context, scanID string) error {
+	if !features.ComplianceEnhancements.Enabled() {
+		return errors.Errorf("Compliance is disabled. Cannot run compliance scan for configuration with ID %s", scanID)
+	}
+
 	// 1. Validate config exists in database
-	// 2. Push request to Sensor
-	panic("implement me")
+	scanConfig, found, err := m.scanSettingDS.GetScanConfiguration(ctx, scanID)
```

Comment:
We should verify the user has write permissions to all the clusters in order to perform this.  Something like https://github.com/stackrox/stackrox/blob/07d1ce838ef367943fcb88d417742a3ac36e8d8f/central/complianceoperator/v2/compliancemanager/manager_impl.go#L98

Reply:
Pulled this out into a separate function so we can reuse it here. Thanks!